msal/src/Microsoft.Identity.Client/MsalConfigurationBuilderExtensions.cs

```diff
@@ -0,0 +1,187 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Extension class to add specific configuration methods to the MsalConfigurationBuilder.
+    /// </summary>
+    public static class MsalConfigurationBuilderExtensions
+    {
+        /// <summary>
+        /// Add an AAD authority to the configuration.
+        /// </summary>
+        /// <param name="builder">The MsalConfigurationBuilder.</param>
+        /// <param name="aadAuthorityAudience">The AAD Authority Audience value to use.</param>
+        /// <returns>MsalConfigurationBuilder</returns>

```

Comment:
This is intenionally just the husk without json parse/serialize and validation.  I want to get a final lock on the public API surface and answer a few of the outstanding questions on some of the value usage before doing validation and serialization and unit testing.

Do we feel there are more settings we want in here that aren't in Shane's spec?